File: openbis_api/source/java/ch/ethz/sis/openbis/generic/asapi/v3/dto/query/get/GetQueryDatabasesOperationResult.java
/*
 * Copyright 2013 ETH Zuerich, CISD
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package ch.ethz.sis.openbis.generic.asapi.v3.dto.query.get;

import java.util.Map;

import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.get.GetObjectsOperationResult;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.query.QueryDatabase;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.query.id.IQueryDatabaseId;
import ch.systemsx.cisd.base.annotation.JsonObject;

/**
 * @author pkupczyk
 */
@JsonObject("as.dto.query.get.GetQueryDatabasesOperationResult")
public class GetQueryDatabasesOperationResult extends GetObjectsOperationResult<IQueryDatabaseId, QueryDatabase>
{

    private static final long serialVersionUID = 1L;

    @SuppressWarnings("unused")
    private GetQueryDatabasesOperationResult()
    {
    }

    public GetQueryDatabasesOperationResult(Map<IQueryDatabaseId, QueryDatabase> objectMap)
    {
        super(objectMap);
    }
}

File: openbis/source/java/ch/systemsx/cisd/openbis/public/resources/api/v3/as/dto/query/get/GetQueryDatabasesOperationResult.js
/**
 * @author pkupczyk
 */
define(['stjs', 'as/dto/common/get/GetObjectsOperationResult'], function (
  stjs,
  GetObjectsOperationResult
) {
  var GetQueryDatabasesOperationResult = function (objectMap) {
    GetObjectsOperationResult.call(this, objectMap)
  }
  stjs.extend(
    GetQueryDatabasesOperationResult,
    GetObjectsOperationResult,
    [GetObjectsOperationResult],
    function (constructor, prototype) {
      prototype['@type'] = 'as.dto.query.get.GetQueryDatabasesOperationResult'
      prototype.getMessage = function () {
        return 'GetQueryDatabasesOperationResult'
      }
    },
    {}
  )
  return GetQueryDatabasesOperationResult
})
